enhancements/sig-policy/98-long-term-compliance-history/README.md

```diff
@@ -0,0 +1,455 @@
+[github.com/stolostron/rbac-api-utils](https://github.com/stolostron/rbac-api-utils/blob/159deac7d398c3470ae730c3fbb8ffbbcca1ac5a/pkg/rbac/rbac.go#L187)
+library to do this.
+
+The following optional query parameters would be accepted on this HTTP endpoint:
+
+- cluster - the name of the cluster to filter on. Multiple can be provided with `?cluster=local-cluster&cluster=managed`
+  syntax.
+- parent_policy - the name of the parent policy (i.e. a `Policy`) of the policy (e.g. a `ConfigurationPolicy`) that
+  generated the compliance event to filter on.
+- policy - the name of the policy (e.g. a `ConfigurationPolicy`) that generated the compliance event to filter on.
+- policy_kind - the kind of the policy (e.g. `ConfigurationPolicy`) that generated the compliance event to filter on.
+- policy_api_group - the `apiGroup` of the policy (e.g. `policy.open-cluster-management.io`) that generated the
+  compliance event to filter on.
+- before - an ISO-8601 timestamp to indicate only compliance events before this time should be shown.
+- after - an ISO-8601 timestamp to indicate only compliance events after this time should be shown.
+- page - the page number in the query. This defaults to 1.
+- per_page - the number of compliance events returned per page. This defaults to 20 and maxes out at 100.
+
+The output would look as follows and be sorted by timestamps in descending order. The content would be filtered based on
+the user's access as described above:
+
+```json
+{
+  "data": [
+    {
+      "cluster": {
+        "name": "cluster1"
+      },
+      "parent_policy": {
+        "name": "policies.etcd-encryption"
+      },
+      "policy": {
+        "apiGroup": "policy.open-cluster-management.io",
+        "kind": "ConfigurationPolicy",
+        "spec": {},
+        "specHash": {}
+      },
+      "event": {
+        "compliance": "NonCompliant",
+        "message": "violation - configmaps [app-data] not found in namespace default",
+        "timestamp": "2023-07-19T18:25:43.511Z",
+        "metadata": {}
+      }
+    },
+    {
+      "cluster": {
+        "name": "cluster2"
+      },
+      "parent_policy": {
+        "name": "policies.etcd-encryption"
+      },
+      "policy": {
+        "apiGroup": "policy.open-cluster-management.io",
+        "kind": "ConfigurationPolicy",
+        "spec": {},
+        "specHash": {}
+      },
+      "event": {
+        "compliance": "NonCompliant",
+        "message": "notification - configmaps [app-data] found as specified in namespace default",
+        "timestamp": "2023-07-19T18:25:41.523Z",
+        "metadata": {}
+      }
+    }
+  ],
+  "metadata": {
+    "page": 1,
+    "per_page": 20,
+    "total": 123
+  }
+}
+```
+
+##### Generating a Report
+
+Another HTTP endpoint would be added at `/api/v1/reports/compliance-events` that would accept a `GET` request with the
+same query parameters as the `/api/v1/compliance-events` HTTP endpoint (excluding pagination), but would generate a CSV
```

Comment:
In the CSV I am guessing the spec would not be included and the message would be reformatted to not contain commas